spyder/plugins/application/widgets/status.py

```diff
@@ -0,0 +1,100 @@
+
+"""
+Status widget for Spyder updates.
+"""
+
+# Standard library imports
+import logging
+
+# Third party imports
+from qtpy.QtCore import Slot
+
+# Local imports
+from spyder import get_versions
+from spyder.api.widgets.status import StatusBarWidget
+from spyder.config.base import _, is_pynsist
+from spyder.plugins.application.widgets.install import (
+    UpdateInstallerDialog, NO_STATUS, DOWNLOADING_INSTALLER, INSTALLING,
+    FINISHED, PENDING, CHECKING, CANCELLED)
+from spyder.utils.icon_manager import ima
+
+logger = logging.getLogger(__name__)
+
+
+class ApplicationUpdateStatus(StatusBarWidget):
+    """Status bar widget for Application update status."""
+    BASE_TOOLTIP = _("Application update status")
+    ID = 'application_update_status'
+
+    def __init__(self, parent):
+
+        self.tooltip = self.BASE_TOOLTIP
+        super().__init__(parent, show_spinner=True)
+
+        # Installation dialog
+        self.installer = UpdateInstallerDialog(self)
+
+        self.sig_clicked.connect(self.show_installation_dialog)
+
+        self.installer.sig_installation_status.connect(
+            self.set_value)
+
+    def set_value(self, value):
+        """Return update installation state."""
+        if value == DOWNLOADING_INSTALLER or value == INSTALLING:
+            self.tooltip = _("Update installation will continue in the "
+                             "background.\n"
+                             "Click here to show the installation "
+                             "dialog again")
+            self.spinner.show()
+            self.spinner.start()
+        elif value == PENDING:
+            self.tooltip = value
+            self.spinner.hide()
+            self.spinner.stop()
+        else:
+            self.tooltip = self.BASE_TOOLTIP
+        self.setVisible(True)
+        self.update_tooltip()
+        value = f"Spyder: {value}"
+        logger.debug(f"Application Update Status: {value}")
+        super().set_value(value)
+
+    def get_tooltip(self):
+        """Reimplementation to get a dynamic tooltip."""
+        return self.tooltip
+
+    def get_icon(self):
+        return ima.icon('spyder_about')
+
+    def start_installation(self):
+        self.installer.start_installation_update()
+
+    def set_status_pending(self):
+        self.set_value(PENDING)
+        self.spinner.hide()
+        self.spinner.stop()
+
+    def set_status_checking(self):
+        self.set_value(CHECKING)
+        self.spinner.show()
+        self.spinner.start()
+
+    def set_no_status(self):
+        self.set_value(NO_STATUS)
+        self.spinner.hide()
+        self.spinner.stop()
+
+    @Slot()
+    def show_installation_dialog(self):
+        """Show installation dialog."""
+        if ((not self.tooltip == self.BASE_TOOLTIP and not
+                self.tooltip == PENDING) and is_pynsist()):
+            self.installer.show()
+        elif ((self.tooltip == PENDING) and is_pynsist()):
+            self.installer.continue_install()
```

Comment:
To add the menu, and also use the actual value set for the validations then this method will become:
```suggestion
    def show_installation_dialog_or_menu(self):
        """Show installation dialog or menu."""
        value = self.value.split(":")[-1].strip()
        if (not self.tooltip == self.BASE_TOOLTIP and not
                value == PENDING and is_pynsist()):
            self.installer.show()
        elif value == PENDING and is_pynsist():
            self.installer.continue_install()
        elif value == NO_STATUS:
            self.menu.clear()
            check_for_updates_action = create_action(
                self,
                text=_("Check for updates..."),
                triggered=self.sig_check_for_updates_requested.emit
            )
            add_actions(self.menu, [check_for_updates_action])
            rect = self.contentsRect()
            os_height = 7 if os.name == 'nt' else 12
            pos = self.mapToGlobal(
                rect.topLeft() + QPoint(-10, -rect.height() - os_height))
            self.menu.popup(pos)
```